prometheus-to-sd/translator/translator.go

```diff
@@ -122,6 +122,63 @@ func DowncaseMetricNames(metricFamilies map[string]*dto.MetricFamily) map[string
 	return result
 }
 
+// FlattenSummaryMetricFamilies flattens summary metric families into two counter metrics,
+// one for the running sum and count, respectively
+func FlattenSummaryMetricFamilies(metricFamilies map[string]*dto.MetricFamily) map[string]*dto.MetricFamily {
+	result := make(map[string]*dto.MetricFamily)
+	for metricName, family := range metricFamilies {
+		switch family.GetType() {
+		case dto.MetricType_SUMMARY:
+			if len(family.Metric) < 1 {
+				glog.Errorf("Summary metric %v does not actually have metric data associated, ignoring", family.Name)
+				continue
+			}
+			summary := family.Metric[0].GetSummary()
```

Comment:
Like that you are going to convert only one sample, which is wrong if summary metric is going to have labels.

Reply:
ah, good catch. will add label support and a test.